src/motion_input_shaping/shaped_axis.py

```diff
@@ -151,6 +215,86 @@ def move_relative_shaped(
             Units.ACCELERATION_MILLIMETRES_PER_SECOND_SQUARED,
         )
 
+    def _move_relative_shaped_stream(
+        self,
+        position: float,
+        unit: Units = Units.NATIVE,
+        wait_until_idle: bool = True,
+        acceleration: float = 0,
+        acceleration_unit: Units = Units.NATIVE,
+    ) -> None:
+        """
+        Input-shaped relative move for the target resonant frequency and damping ratio.
+
+        :param position: The amount to move.
+        :param unit: The units for the position value.
+        :param wait_until_idle: If true the command will hang until the device reaches idle state.
+        :param acceleration: The acceleration for the move.
+        :param acceleration_unit: The units for the acceleration value.
+        """
+        # Convert all to values to the same units
+        position_native = super().settings.convert_to_native_units("pos", position, unit)
+        accel_native = super().settings.convert_to_native_units(
+            "accel", acceleration, acceleration_unit
+        )
+
+        if acceleration == 0:  # Get the acceleration if it wasn't specified
+            accel_native = super().settings.get("accel", Units.NATIVE)
+
+        position_mm = super().settings.convert_from_native_units(
+            "pos", position_native, Units.LENGTH_MILLIMETRES
+        )
+        accel_mm = super().settings.convert_from_native_units(
+            "accel", accel_native, Units.ACCELERATION_MILLIMETRES_PER_SECOND_SQUARED
+        )
+
+        start_position = super().get_position(Units.LENGTH_MILLIMETRES)
+
+        stream_segments = self._shaper_stream.shape_trapezoidal_motion(
+            position_mm,
+            accel_mm,
+            accel_mm,
+            self.get_max_speed_limit(Units.VELOCITY_MILLIMETRES_PER_SECOND),
+        )
+        self.stream.disable()
+        self.stream.setup_live(self.axis_number)
+        self.stream.cork()
+        for segment in stream_segments:
+            # Set acceleration making sure it is greater than zero by comparing 1 native accel unit
```

Comment:
I think he's trying to make sure that accel could never round down to 0 in Zaber units if it was very low (0.01m/s/s). I do think this approach is most robust, but it also might be excessive? Never hurts I guess.

Reply:
I did this for robustness and to prevent rounding down to 0 as Graham suggested. There are scenarios where this can happen if the accel is set very low to begin with or if the shaper is configured in a way that results in a segment of very low acceleration.